src/solver/algorithm/strong/puzzle.rs

```diff
@@ -106,13 +244,576 @@ where
         .context("Failed to select solution set database table.")?;
 
     Ok(db)
+
+}
+*/
+
+// THIS IS ONLY FOR TESTING PURPOSES
+struct TestDB {
```

Comment:
This should use one of the actual DB implementations (like `VolatileDB`, which is pretty much the same thing). Independently of that:

When you do anything only for testing purposes, it should be marked `#[cfg(test)]`. You can achieve this by marking the item with this directive,
```
#[cfg(test)]
struct TestDB { ... }
```
or by placing the whole thing in the `tests` module (which is already marked as `cfg(test)`.

This tells the compiler to not include it in release builds.